From 2deb46a280c350a096890f1e3977b97a11f55a5d Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:59 -0400
Subject: [fs] btrfs: fix nossd and ssd_spread mount option regression

Message-id: <1407179109-13714-413-git-send-email-sandeen@redhat.com>
Patchwork-id: 87283
O-Subject: [RHEL7.1 PATCH 412/422] btrfs: fix nossd and ssd_spread mount option regression
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

commit 2aa06a35d06a34b3109bdbf1d653de1695dc8f12
Author: Eric Sandeen <sandeen@redhat.com>
Date:   Fri Jun 27 16:50:31 2014 -0500

    btrfs: fix nossd and ssd_spread mount option regression

    The commit

    0780253 btrfs: Cleanup the btrfs_parse_options for remount.

    broke ssd options quite badly; it stopped making ssd_spread
    imply ssd, and it made "nossd" unsettable.

    Put things back at least as well as they were before
    (though ssd mount option handling is still pretty odd:
    # mount -o "nossd,ssd_spread" works?)

    Reported-by: Roman Mamedov <rm@romanrm.net>
    Signed-off-by: Eric Sandeen <sandeen@redhat.com>
    Signed-off-by: Chris Mason <clm@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index e2f3b01..9b35078 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -522,9 +522,10 @@ int btrfs_parse_options(struct btrfs_root *root, char *options)
   case Opt_ssd_spread:
    btrfs_set_and_info(root, SSD_SPREAD,
         "use spread ssd allocation scheme");
+   btrfs_set_opt(info->mount_opt, SSD);
    break;
   case Opt_nossd:
-   btrfs_clear_and_info(root, NOSSD,
+   btrfs_set_and_info(root, NOSSD,
           "not using ssd allocation scheme");
    btrfs_clear_opt(info->mount_opt, SSD);
    break;
-- 
1.7.1